buildchain/buildchain/packaging.py

```diff
@@ -52,7 +52,7 @@ def _list_packages_to_build(
 def _list_packages_to_download(
     package_versions: Tuple[versions.PackageVersion, ...],
     packages_to_build: List[str]
-) -> Dict[str,str]:
+) -> Mapping[str, Optional[str]]:
```

Comment:
Why the change here? (and BTW, why the change in general: a commit message would be welcome).

IIRC, the good practice is to use "precise" type for return type and "abstract" one for input.

See https://docs.python.org/3/library/typing.html#typing.Dict:

> Useful for annotating return types. To annotate arguments it is preferred to use an abstract collection type such as Mapping.